File: demo2/conduit_companies.py
from typing import Any
from urllib.parse import urljoin

import requests

CONDUIT_API_URL = 'https://api.getconduit.app'
CONDUIT_API_KEY = ' place your api key here '


def get_companies() -> list[dict[str, Any]]:
    """
    Returns list of companies
    :return: list of users
    """
    res = _request('link/company/?include_connections=true', method='GET')
    return res.json()


def create_company(
        company_id: str,
        name: str = None,
        page_enabled: bool = True,
) -> dict[str, Any]:
    """
    Creates a new company with given name and email.
    :param company_id: company guid(unique within API consumer)
    :param name: name of the company
    :param page_enabled: Enabling company credentials page
    """
    res = _request('link/company/', params=_get_company_params(company_id, name, page_enabled))
    return res.json()


def get_user_token(user_guid: str) -> str:
    """
    Returns user's access token.
    :param user_guid: user's guid
    :return: token
    """
    res = _request('link/auth/token/', params={'guid': user_guid})
    return res.json()['token']


def _get_company_params(company_id: str, name: str, page_enabled: bool) -> dict[str, str]:
    params = {
        'id': company_id,
    }
    if name is not None:
        params['name'] = name
    if page_enabled is not None:
        params['page_enabled'] = page_enabled

    return params


def _request(path: str, method: str = 'POST', params: dict[str, Any] = None) -> Any:
    headers = {
        'accept': 'application/json',
        'Authorization': f'Bearer {CONDUIT_API_KEY}',
    }
    res = requests.request(method, urljoin(CONDUIT_API_URL, path), headers=headers, json=params)
    res.raise_for_status()

    return res

File: demo2_nodejs/conduit_companies.mjs
import axios from 'axios'

const CONDUIT_API_URL = 'https://api.getconduit.app'
const CONDUIT_API_KEY = ' place your api key here '

const axiosInstance = axios.create({
    baseURL: CONDUIT_API_URL,
    headers: {accept: 'application/json'},
})

async function appRequest(endpoint, method = 'POST', data = {}) {
    return axiosInstance.request({
        url: endpoint,
        method: method,
        data: data,
        headers: {Authorization: `Bearer ${CONDUIT_API_KEY}`},
    })
}

async function companyRequest(endpoint, token, method = 'GET', params = {}) {
    return axiosInstance.request({
        url: endpoint,
        method: method,
        params: {token: token, ...params},
    })
}

/**
 * @typedef {{name: String, id: Number, native_id: String}} Account
 * @typedef {{name: String, id: Number, created_at: String, accounts: Account[]}} Credentials
 * @typedef {{id: String, name: String, credentials: Credentials[]}} Connection
 * @typedef {{
 * id: String,
 * name: String,
 * created_at: String,
 * link_page: Object.<{url: String, enabled: Boolean}>,
 * api_token: Object.<{token: String, expires_at: Number}>,
 * connections: Connection[]}} Company
 */

/**
 *
 * Gets a list of companies.
 * @constructor
 * @returns {Promise.<Company[]>} list of companies
 */
async function getCompanies() {
    let response = await appRequest('link/company/?include_connections=true', 'GET')
    return response.data
}

/**
 *
 * Creates company.
 * @constructor
 * @param {String}  companyId - company UUID.
 * @param {String}  name - company name.
 * @param {Boolean}  pageEnabled - initial link page state.
 * @returns {Promise.<Object.<{id: String, name:String, page_enabled: Boolean}>>} created company.
 */
async function createCompany(companyId, name, pageEnabled) {
    let response = await appRequest('link/company/', 'POST', {
        id: companyId,
        name: name,
        page_enabled: pageEnabled
    })
    return response.data
}

/**
 *
 * Creates company.
 * @constructor
 * @param {String}  companyToken - a company token.
 * @param {String}  integration - integration id.
 * @returns {Promise.<String>} Connect URL.
 */
async function getCompanyConnectIntegrationURL(companyToken, integration) {
    let response = await companyRequest(`link/credentials/connect/${integration}/`, companyToken, 'GET')
    return response.data.url
}


export { createCompany, getCompanies, getCompanyConnectIntegrationURL }
